test/org/zaproxy/zap/authentication/UsernamePasswordAuthenticationCredentialsUnitTest.java

```diff
@@ -0,0 +1,128 @@
+    }
+
+    @Test
+    public void shouldBeConfiguredIfUsernameAndPasswordAreNotNull() throws Exception {
+        // Given/When
+        boolean isConfigured = usernamePasswordAuthenticationCredentials.isConfigured();
+
+        // Then
+        assertThat(isConfigured, is(true));
+    }
+
+    @Test
+    public void shouldNotBeConfiguredIfUsernameAndPasswordAreNull() throws Exception {
+        // Given/When
+        boolean isConfigured = notConfiguredInstance.isConfigured();
+
+        // Then
+        assertThat(isConfigured, is(false));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldThrowExceptionWhileEncodeWithFieldSeparator() throws Exception {
+        // Given
+        String fieldSeparator = "~";
+        usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+
+        // When
+        usernamePasswordAuthenticationCredentials.encode(fieldSeparator);
+
+        // Then throw IllegalArgumentException
+    }
+
+    @Test
+    public void shouldEncodeMethodReturnNullPatternIfUsernameIsNull() throws Exception {
+        // Given
+        String nullPattern = "AA==";
+        username = null;
+        password = "something";
+        String stringSeparator = "|";
+        usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+
+        // When
+        String encodedResult = usernamePasswordAuthenticationCredentials.encode(stringSeparator);
+
+        // Then
+        assertThat(encodedResult, is(nullPattern));
+    }
+
+    @Test
+    public void shouldEncodeUsernameAndPasswordWithTheCorrectFieldSeparator() throws Exception {
+        // Given
+        List<String> someCorrectSeparators = Arrays.asList("-", "|", "/", "\\", "+");
+
+        // When/Then
+        for (String correctSeparator : someCorrectSeparators) {
+            String encodedUsernamePassword = usernamePasswordAuthenticationCredentials.encode(correctSeparator);
+
+            assertThat(String.format("Failed to encode with '%s'", correctSeparator), encodedUsernamePassword, notNullValue());
+            assertThat(
+                    String.format("Failed to properly encode with '%s'", correctSeparator),
+                    encodedUsernamePassword,
+                    is("bXlVc2Vy~bXlQYXNz~"));
+        }
+    }
+
+    @Test
+    public void shouldApiResponseRepresentationReturnApiResponseWithValidNameAndJsonFormat() throws Exception {
+        // Given/When
+        ApiResponse apiResponse = usernamePasswordAuthenticationCredentials.getApiResponseRepresentation();
+        JSON jsonRepresentation = apiResponse.toJSON();
+
+        // Then
+        assertThat(apiResponse, notNullValue());
+        assertThat(apiResponse.getName(), equalToIgnoringCase("credentials"));
+        assertThat(jsonRepresentation.toString(), allOf(containsString("myPass"), containsString("myUser"), containsString("type")));
```

Comment:
- Should check "UsernamePasswordAuthenticationCredentials" instead of "type"? (or also add the former);
 - The "myPass"  and "myUser" should be extracted to a constant (and the corresponding variables initialised with those), less changes if one needs to be changed.